testutil/checkers.go

```diff
@@ -83,3 +83,61 @@ func (c *containsChecker) Check(params []interface{}, names []string) (result bo
 		return false, fmt.Sprintf("%T is not a supported container", container)
 	}
 }
+
+type deepContainsChecker struct {
+	*check.CheckerInfo
+}
+
+// DeepContains is a Checker that looks for a elem in a container using
+// DeepEqual.  The elem can be any object. The container can be an array, slice
+// or string.
+var DeepContains check.Checker = &deepContainsChecker{
+	&check.CheckerInfo{Name: "DeepContains", Params: []string{"container", "elem"}},
+}
+
+func (c *deepContainsChecker) Check(params []interface{}, names []string) (result bool, error string) {
+	defer func() {
+		if v := recover(); v != nil {
```

Comment:
What logic below can panic in an expected way?  In general if we don't know, then we should not recover from it.  The case in Contains above is mimicking what Equals does, and that is trying to not break down badly on improper equalities, which when testing is actually not exceptional at all, but rather what the test is indeed trying to do.

Below DeepEquals on panic, though, and so if we our logic is broken, it should likely panic for real because we have broken code we do not understand.  I'd also suggest tightening up Contains a little bit, by moving the recovering logic to a function that is meant to overview the equality only.